pkg/gadgets/snapshot/socket/tracer/tracer_test.go

```diff
@@ -0,0 +1,176 @@
+// Copyright 2022-2023 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build linux
+// +build linux
+
+package tracer
+
+import (
+	"fmt"
+	"net"
+	"testing"
+
+	utilstest "github.com/inspektor-gadget/inspektor-gadget/internal/test"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets/snapshot/socket/types"
+	eventtypes "github.com/inspektor-gadget/inspektor-gadget/pkg/types"
+	"github.com/stretchr/testify/require"
+)
+
+func TestSocketTracerCreate(t *testing.T) {
+	t.Parallel()
+
+	utilstest.RequireRoot(t)
+
+	tracer, err := NewTracer(types.ALL)
+	require.Nil(t, err, "creating tracer: %v", err)
+
+	tracer.CloseIters()
+}
+
+type extraInformation struct {
+	addr    string
+	port    uint16
+	netNsID uint64
+}
```

Comment:
The original idea of having this "extra" structure was to be able to pass information that is dynamic, for instance, the port a socket uses when it's bound to `0`. In this case, `addr` and `port` are known at test creation, and `netNsID` is available on `runner.Info`. I used the following to make it work without this structure:

```diff
diff --git pkg/gadgets/snapshot/socket/tracer/tracer_test.go pkg/gadgets/snapshot/socket/tracer/tracer_test.go
index 41b067d4..6c713395 100644
--- pkg/gadgets/snapshot/socket/tracer/tracer_test.go
+++ pkg/gadgets/snapshot/socket/tracer/tracer_test.go
@@ -39,18 +39,12 @@ func TestSocketTracerCreate(t *testing.T) {
        tracer.CloseIters()
 }
 
-type extraInformation struct {
-       addr    string
-       port    uint16
-       netNsID uint64
-}
-
 type testCase struct {
        name          string
        proto         types.Proto
        addr          string
        port          int
-       expectedEvent func(info *utilstest.RunnerInfo, e extraInformation) *types.Event
+       expectedEvent func(info *utilstest.RunnerInfo, _ any) *types.Event
        socketCreator func(addr string, port int) error
 }
 
@@ -65,17 +59,17 @@ func TestSnapshotSocket(t *testing.T) {
                        proto: types.TCP,
                        addr:  "127.0.0.1",
                        port:  8082,
-                       expectedEvent: func(info *utilstest.RunnerInfo, e extraInformation) *types.Event {
+                       expectedEvent: func(info *utilstest.RunnerInfo, _ any) *types.Event {
                                return &types.Event{
                                        Event:       eventtypes.Event{Type: eventtypes.NORMAL},
-                                       WithNetNsID: eventtypes.WithNetNsID{NetNsID: e.netNsID},
+                                       WithNetNsID: eventtypes.WithNetNsID{NetNsID: info.NetworkNsID},
                                        Protocol:    "TCP",
                                        Status:      "LISTEN",
                                        SrcEndpoint: eventtypes.L4Endpoint{
                                                L3Endpoint: eventtypes.L3Endpoint{
-                                                       Addr: e.addr,
+                                                       Addr: "127.0.0.1",
                                                },
-                                               Port: e.port,
+                                               Port: 8082,
                                        },
                                        DstEndpoint: eventtypes.L4Endpoint{
                                                L3Endpoint: eventtypes.L3Endpoint{
@@ -100,17 +94,17 @@ func TestSnapshotSocket(t *testing.T) {
                        proto: types.UDP,
                        addr:  "127.0.0.1",
                        port:  8082,
-                       expectedEvent: func(info *utilstest.RunnerInfo, e extraInformation) *types.Event {
+                       expectedEvent: func(info *utilstest.RunnerInfo, _ any) *types.Event {
                                return &types.Event{
                                        Event:       eventtypes.Event{Type: eventtypes.NORMAL},
-                                       WithNetNsID: eventtypes.WithNetNsID{NetNsID: e.netNsID},
+                                       WithNetNsID: eventtypes.WithNetNsID{NetNsID: info.NetworkNsID},
                                        Protocol:    "UDP",
                                        Status:      "INACTIVE",
                                        SrcEndpoint: eventtypes.L4Endpoint{
                                                L3Endpoint: eventtypes.L3Endpoint{
-                                                       Addr: e.addr,
+                                                       Addr: "127.0.0.1",
                                                },
-                                               Port: e.port,
+                                               Port: 8082,
                                        },
                                        DstEndpoint: eventtypes.L4Endpoint{
                                                L3Endpoint: eventtypes.L3Endpoint{
@@ -151,11 +145,6 @@ func TestSnapshotSocket(t *testing.T) {
                        evs, err := tracer.RunCollector(uint32(runner.Info.Tid), "", "", "")
                        require.ErrorIsf(t, nil, err, "running collector: %v", err)
 
-                       type extra struct {
-                               addr string
-                               port uint16
-                       }
-
                        events := make([]types.Event, len(evs))
                        for i, ev := range evs {
                                events[i] = *ev
@@ -165,12 +154,7 @@ func TestSnapshotSocket(t *testing.T) {
                                events[i].InodeNumber = 0
                        }
 
-                       utilstest.ExpectAtLeastOneEvent(c.expectedEvent)(t, nil, extraInformation{
-                               addr:    c.addr,
-                               port:    uint16(c.port),
-                               netNsID: runner.Info.NetworkNsID,
-                       },
-                               events)
+                       utilstest.ExpectAtLeastOneEvent(c.expectedEvent)(t, runner.Info, nil, events)
                })
        }
 }

```